rgw/v2/tests/curl/reusable.py

```diff
@@ -0,0 +1,226 @@
+    upload_object_status = utils.exec_shell_cmd(command)
+    if upload_object_status is False:
+        raise TestExecError("object upload failed")
+    log.info(f"object {s3_object_name} uploaded")
+    return True
+
+
+def download_object(
+    curl_auth, bucket_name, s3_object_name, TEST_DATA_PATH, s3_object_path
+):
+    """
+    download object using curl
+    ex: curl -X GET http://10.0.209.142:80/bkt1/obj1 -o /home/cephuser/out_file_name
+    Args:
+        curl_auth(CURL): CURL object instantiated with access details and endpoint
+        bucket_name(str): Name of the bucket to be created
+        s3_object_name(str): name of the s3 object
+        s3_object_path(str): path of the s3 object on the local node
+        TEST_DATA_PATH(str): test data path where objects created are stored on ceph-qe-scripts local repo
+    """
+    log.info(f"s3 object name to download: {s3_object_name}")
+    s3_object_download_name = s3_object_name + "." + "download"
+    s3_object_download_path = os.path.join(TEST_DATA_PATH, s3_object_download_name)
+    headers = {
+        "x-amz-content-sha256": "UNSIGNED-PAYLOAD",
+    }
+    command = curl_auth.command(
+        http_method="GET",
+        headers=headers,
+        output_file=s3_object_download_path,
+        url_suffix=f"{bucket_name}/{s3_object_name}",
+    )
+    upload_object_status = utils.exec_shell_cmd(command)
+    if upload_object_status is False:
+        raise TestExecError("object download failed")
+    log.info(f"object {s3_object_name} downloaded")
+
+    s3_object_downloaded_md5 = utils.get_md5(s3_object_download_path)
+    s3_object_uploaded_md5 = utils.get_md5(s3_object_path)
+    log.info(f"s3_object_downloaded_md5: {s3_object_downloaded_md5}")
+    log.info(f"s3_object_uploaded_md5: {s3_object_uploaded_md5}")
+    if str(s3_object_uploaded_md5) == str(s3_object_downloaded_md5):
+        log.info("md5 match")
+        utils.exec_shell_cmd(f"rm -rf {s3_object_download_path}")
+    else:
+        raise TestExecError("md5 mismatch")
+
+
+def delete_object(curl_auth, bucket_name, s3_object_name):
+    """
+    delete object using curl
+    ex: curl -X DELETE http://10.0.209.142:80/bkt1/obj1
+    Args:
+        curl_auth(CURL): CURL object instantiated with access details and endpoint
+        bucket_name(str): Name of the bucket to be created
+        s3_object_name(str): name of the s3 object
+    """
+    log.info(f"s3 object to delete: {s3_object_name}")
+    headers = {
+        "x-amz-content-sha256": "UNSIGNED-PAYLOAD",
+    }
+    command = curl_auth.command(
+        http_method="DELETE",
+        headers=headers,
+        url_suffix=f"{bucket_name}/{s3_object_name}",
+    )
+    delete_object_status = utils.exec_shell_cmd(command)
+    if delete_object_status is False:
+        raise TestExecError("object deletion failed")
+    log.info(f"object {s3_object_name} deleted")
+    return True
+
+
+def delete_bucket(curl_auth, bucket_name):
+    """
+    delete object using curl
```

Comment:
please correct the statement,should be "delete bucket using curl"

Reply:
@ckulal done, please review again!